src/test/ui/associated-const/defaults-cyclic-fail.rs

```diff
@@ -1,9 +1,9 @@
 // build-fail
+//~^ ERROR cycle detected when normalizing `<() as Tr>::A`
 
 // Cyclic assoc. const defaults don't error unless *used*
 trait Tr {
     const A: u8 = Self::B;
-    //~^ ERROR cycle detected when const-evaluating + checking `Tr::A`
 
     const B: u8 = Self::A;
 }
```

Comment:
This probably needs to be improved.

Might try to either add a valid  `DefId` to every used `ParamEnv` or try to get a contained def id in the `Ty` or `Const`.
Not yet sure what works best

Reply:
Neither is a great idea IMO, because they would impair caching.